### ISSUE TITLE
0.10.5: setting _daemon in filter definition does not work

### ISSUE THREAD

**Issue by tjikkun** [fail2ban/fail2ban#2650]
_We will be very grateful, if your problem was described as completely as possible,
enclosing excerpts from logs (if possible within DEBUG mode, if no errors evident
within INFO mode), and configuration in particular of effected relevant settings
(e.g., with ` fail2ban-client -d | grep 'affected-jail-name' ` for a particular
jail troubleshooting).
Thank you in advance for the details, because such issues like "It does not work" 
alone could not help to resolve anything!
Thanks! (remove this paragraph and other comments upon reading)_

### Environment:

- Fail2Ban version (including any possible distribution suffixes): 0.10.5-2.el7
- OS, including release name/version: CentOS Linux release 7.7.1908 (Core)
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:

Starting from 0.10.5, it seems _daemon setting in filter definition does not work anymore.
In filter.d we have for a filter:
```
[Definition]

_daemon = mail smtpd

failregex = ^%(__prefix_line)s\[[a-f0-9-]+\] <SNIP>
```
With 0.10.5 this translates to (fail2ban-client -d output):
`['set', 'mailfilter', 'addfailregex', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel:\\s?\\[ *\\d+\\.\\d+\\]:?\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?\\S*(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?\\S*(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?\\[[a-f0-9-]+\\] <SNIP>`
While with 0.10.4 this translated to:
`['set', 'mailfilter', 'addfailregex', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?mail smtpd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?mail smtpd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?\\[[a-f0-9-]+\\] <SNIP>`
As you can see, with 0.10.4 daemon was overriden, but not with 0.10.5

#### Steps to reproduce
* add `_daemon = find this string` definition to filter file
* run `fail2ban-client -d | grep 'find this string'`

#### Expected behavior
grep finds a match

#### Observed behavior
grep doesn't find a match

**Comment by sebres** [comment]
Good catch... Confirmed.
Thanks!

**Comment by sebres** [comment]
The regression was introduced in 62b1712d223269b6201366e55c796a15a2f2211d (amend to PR #2387), so default values are directly substituted if value of `__prefix_line` gets substituted in related section `lt_*`:
https://github.com/fail2ban/fail2ban/blob/62b1712d223269b6201366e55c796a15a2f2211d/config/filter.d/common.conf#L61

Fixed now in 42714d0849cff8a1502e0fabfed376e40187fca8 (interpolation type of default values in `common.conf` switched to RTM substitution, working across sections).
Until it gets released you can replace configs e. g. [filter.d/common.conf](https://github.com/fail2ban/fail2ban/blob/42714d0849cff8a1502e0fabfed376e40187fca8/config/filter.d/common.conf)

**Comment by tjikkun** [comment]
I just tried this on 0.11.1 but it doesn't seem to work

**Comment by sebres** [comment]
what exactly did you try?

**Comment by tjikkun** [comment]
Well I updated my installation, and it broke, then i ran the "Steps to reproduce" and grep didn't find anything. So for me, it looks like it is the same issue as with 0.10.5
>  add _daemon = find this string definition to filter file
>  run fail2ban-client -d | grep 'find this string'

**Comment by sebres** [comment]
Well installation of 0.11.1 would do nothing, because the fix 42714d0849cff8a1502e0fabfed376e40187fca8 is not yet released (neither in 0.10 nor in 0.11 branches).
Did you update the config manually as I wrote above?

**Comment by tjikkun** [comment]
Oh sorry! I thought 0.11.1 would contain the fix. With updated config things are okay, but I wanted to see how things were with released fix. My bad!

**Comment by sebres** [comment]
No problem... 
0.11.1 and 0.10.5 are released together at same point, so the fix would target both also (0.11.2 and 0.10.6).
